### ISSUE TITLE
using splide in nuxtjs

### ISSUE THREAD

**Issue by amirhkhadem** [Splidejs/vue-splide#27]
Has anyone tried to use a Vue solution in Nuxt as a plugin or a module?

I'm having lots of challenges trying to do this the right way!

I'm trying to import Splide Vue slider from NPM into Nuxt and after you install it via NPM, there seems to be a challenge on how to import it into a splide.js file in plugins then try to address it in nuxtconfig.js as a plugin. I get all sorts of errors.

here are my files:

nuxt.config.js

```
modules: [
    // https://go.nuxtjs.dev/bootstrap
    '@nuxtjs/style-resources',
    '@nuxtjs/bulma', 
    '@splidejs/splide'
    
  ],
```
splide.vue (in nuxt components)

```
<template>
  <splide :options="options">
        <splide-slide>
            <h1>boo</h1>
        </splide-slide>
        <splide-slide>
            <h1>boo2</h1>
        </splide-slide>
  </splide>
</template>

<script>
  export default {
      data() {
          return {
              options: {
                rewind : true,
                  width  : 800,
                  perPage: 1,
                  gap    : '1rem',
              },
          };
      },
  }
</script>
```
splide.js (in plugins)

```
import Vue from 'vue';
import VueSplide from 'vue-splide';

Vue.use( VueSplide );
new Vue( {
    el    : '#app',
    render: h => h( App ),
  } );
```
this was the link on the splide site: splide instructions

my most recent error is "window is not defined"

**Comment by laraketai** [comment]
plugin:
`import Vue from 'vue';
import App from './App';
import VueSplide from '@splidejs/vue-splide';`


nuxt.config.js:
remove '@splidejs/splide' from modules

add your plugin

plugins: [
    { src: '@/plugins/splideplugin' },
  ],

**Comment by amirhkhadem** [comment]
Thanks @laraketai I solved it in stackoverflow, here is the link: [https://stackoverflow.com/questions/65545747/splidejs-in-nuxt-vue](url)

**Comment by kissu** [comment]
Not sure about the current state of the project (no `Releases tab`) but looking at this [SO answer](https://stackoverflow.com/a/65546429/8816585), the solution is clearly not reproducible and throws a lot of errors regarding Vue3 Composition API as it can be seen here.

![image](https://user-images.githubusercontent.com/5133074/143970533-f4ce43d2-4e12-490f-a891-424537f718a4.png)

Not sure if the version is somehow not compatible with Nuxt2 (using Vue2) as of today, but downgrading `@splidejs/vue-splide` to `0.3.5` solved the issue and is working fine with Nuxt2.

@NaotoshiFujita can you confirm that backwards compatibility was lost somewhere? Or do we need to write it in another way here?

**Comment by NaotoshiFujita** [comment]
All newer versions than v0.3.5 work only for Vue3 and they don't have backwards compatibility.
v0.3.5 works with Vue2, but it has the SSR issue.

**Comment by kissu** [comment]
@NaotoshiFujita you're talking about [this one](https://github.com/Splidejs/vue-splide/issues/15)?

**Comment by NaotoshiFujita** [comment]
Right, the window reference error. You have to run the version as a client mode if you really need to use it in Vue2.

**Comment by kissu** [comment]
Yep, using [`client-only`](https://nuxtjs.org/docs/features/nuxt-components#the-client-only-component) is enough to fix the [DOM mismatch issue](https://stackoverflow.com/a/67978474/8816585). Thanks for the details! 🙏🏼

**Comment by NaotoshiFujita** [comment]
😎👍

**Comment by modbender** [comment]
Created it
https://github.com/modbender/nuxt-splide
